### ISSUE TITLE
Stepper update to drop length/index props

### ISSUE THREAD

**Issue by endigo9740** [Brain-Bones/skeleton#222]
We had a great suggestion on the Svelte support section that explained how we can build the child count (length prop) and child index values (index props) into the components to be automatically generated and maintained.

Here's a basic proof of concept:
https://svelte.dev/repl/e2ca4f024dfa482f9715edab4f67b677?version=3.50.1

Here's a more advanced version that allows for children to be added/removed dynamically:
https://svelte.dev/repl/d5d784109e2145dd8f41906e16bfb89a?version=3.50.1

Unfortunately these seem subject to the same Vite HMR (hot module reload) issue that our Code Block is. When code is modified in the Step component the index values get multiplied. However, if we can work around this, it would make the Stepper MUCH easier for end users to implement.

**Comment by Sarenor** [comment]
Trying to take care of this :)

**Comment by endigo9740** [comment]
Assigned!

**Comment by endigo9740** [comment]
See also: https://github.com/Brain-Bones/skeleton/issues/332

**Comment by endigo9740** [comment]
@Sarenor So I spent some time testing this today. Unfortunately, as cool as the "self register" method is per the REPLs above, this method is just too slow for a real world use case. Even stripped down to the basics, there's visually noticeable delay while the setup process occurs.

![stepper-test](https://user-images.githubusercontent.com/1509726/194772920-9cb5e6f8-9d7a-44df-82c4-1a8162f0026f.gif)

Plus, in additional to the visual delay, you also get issues with the HMR of Vite where the child indexes start duplicating. This requires a hard refresh on the page to correct. Here's what that looks like. Note I have 3 children, but 6 items registered in the `$children` store. If I continued to make changes in the Step component these would continue to duplicate forever!

![Screen Shot 2022-10-09 at 1 01 03 PM](https://user-images.githubusercontent.com/1509726/194772912-15e0ca0a-da02-4c86-82dc-5402b2b0981f.png)

Given this, I think our current implementation is going to be the best we can do for the moment. At least until another technique for handling this is found. To be honest, it's annoying having to supply values manually, but it's not the end of the world. If people want to use this component, they will.

I think with this I'm going to close out this issue and the PR. Though they'll remain archived if we need to refer to them in the future. Thanks for your help on testing this one and sorry this hasn't worked out. Sometimes that's how it goes!
